domain/src/test/java/org/fao/geonet/domain/ISODateTest.java

```diff
@@ -448,4 +453,34 @@ public void testSub() throws Exception {
         org.junit.Assert.assertEquals(-60, isoDate2.timeDifferenceInSeconds(isoDate1));
     }
 
+
+    @Test
+    public void testZ() throws Exception {
+        ISODate isoDate = new ISODate("2019-06-01T00:00Z");
+        Instant instant = isoDate.toDate().toInstant().truncatedTo(ChronoUnit.SECONDS);
+
+        ZonedDateTime expectedDateTime =
+            ZonedDateTime.of(
+                2019, 6, 1, 0, 0, 0, 0, ZoneId.of("Z").normalized());
+        Instant expected = expectedDateTime.toInstant().truncatedTo(ChronoUnit.SECONDS);
+        assertEquals( "Z", expected, instant );
+
+        ISODate pstDate = new ISODate("2019-06-01T00:00-07:00");
+        instant = pstDate.toDate().toInstant().truncatedTo(ChronoUnit.SECONDS);
+
+        ZonedDateTime pstDateTime =
+            ZonedDateTime.of(
+                2019, 6, 1, 0, 0, 0, 0, ZoneId.of("America/Vancouver").normalized());
+        expected = pstDateTime.toInstant().truncatedTo(ChronoUnit.SECONDS);
+        assertEquals( "PST", expected, instant );
+
+        ISODate estDate = new ISODate("2019-06-01T00:00-04:00");
```

Comment:
I see that you are using "-04:00" in the test and comparing it to "America/Toronto"
I believe this test will fail roughly half the year when it switches to EDT